profiler/src/main/java/com/splunk/opentelemetry/profiler/ProfilerEnvVarsConfiguration.java

```diff
@@ -52,44 +52,55 @@ public class Configuration {
   static final String CONFIG_KEY_TRACING_STACKS_ONLY = "splunk.profiler.tracing.stacks.only";
   static final String CONFIG_KEY_STACK_DEPTH = "splunk.profiler.max.stack.depth";
 
-  private static final boolean HAS_OBJECT_ALLOCATION_SAMPLE_EVENT = getJavaVersion() >= 16;
-
   private static final String DEFAULT_PROFILER_DIRECTORY = System.getProperty("java.io.tmpdir");
   private static final Duration DEFAULT_RECORDING_DURATION = Duration.ofSeconds(20);
   private static final Duration DEFAULT_CALL_STACK_INTERVAL = Duration.ofSeconds(10);
 
-  public static void log(ConfigProperties config) {
+  private final ConfigProperties config;
+
+  public ProfilerEnvVarsConfiguration(ConfigProperties config) {
+    this.config = config;
+  }
+
+  @Override
+  public void log() {
```

Comment:
Can this implementation be shared between the two? Seems like it's effectively duplicated...and if they're only calling interface methods should be possible...